h/static/scripts/annotator/anchoring/pdf.coffee

```diff
@@ -0,0 +1,224 @@
+seek = require('dom-seek')
+
+Annotator = require('annotator')
+xpathRange = Annotator.Range
+
+html = require('./html')
+{TextPositionAnchor, TextQuoteAnchor} = require('./types')
+
+
+getSiblingIndex = (node) ->
+  siblings = Array.prototype.slice.call(node.parentNode.childNodes)
+  return siblings.indexOf(node)
+
+
+getNodeTextLayer = (node) ->
+  until node.classList?.contains('page')
+    node = node.parentNode
+  return node.getElementsByClassName('textLayer')[0]
+
+
+getPage = (pageIndex) ->
+  return PDFViewerApplication.pdfViewer.pages[pageIndex]
+
+
+getPageTextContent = (pageIndex, cache) ->
+  if cache[pageIndex]?
+    return Promise.resolve(cache[pageIndex])
+  else
+    joinItems = ({items}) ->
+      nonEmpty = (item.str for item in items when /\S/.test(item.str))
+      textContent = nonEmpty.join('')
+      cache[pageIndex] = textContent
+      return textContent
+
+    return PDFViewerApplication.pdfViewer.getPageTextContent(pageIndex)
```

Comment:
It might be worth noting here that `pdfViewer.getPageTextContent` isn't obviously public API for PDF.js.